### PR TITLE
Reintegrate react-ftm & update to Blueprint 4

### LINKED ISSUE #2146: Adopt Dart Sass as a replacement for node sass
node-sass is now deprecated and we should update to use dart-sass. However, dart-sass is currently incompatible with. the ARM based M1 macs.

Once dart is supported on the M1 architecture, update to use it and deploy with Aleph.

### PR DESCRIPTION
**Pull request by tillprochaska**
This PR includes two major changes: It moves the `react-ftm` codebase back into the main `aleph` repository. And it includes an upgrade to Blueprint v4. While these are separate tasks, I decided to tackle them in a single PR. Both tasks require a lot of manual UI testing to ensure no regressions are introduced, and it’s more convenient to do that once for both tasks.

This PR includes changes to a lot of files, which makes it a little bit hard to review. Many of those changes are automated (using automated migration tooling or running semi-automated find-and-replace). Rather than taking a look at all of the changes at once, it probably makes sense to look at individual commits.

Here’s a summary of the changes. The commit messages contain additional details.

### 1) react-ftm
* Copy-pasted all source files into `ui/src/react-ftm`, leaving the internal directory structure intact, and adjusted import paths where necessary. Also added all react-ftm’s npm dependencies to Aleph’s `package.json`.
* Fixed a couple of TypeScript-related errors and installed missing type definitions.
* Set up tests with Jest and React Testing Library and made sure the few tests we have in react-ftm run successfully.
* Merged all translations from react-ftm into Aleph’s UI translations.

### 2) Blueprint 4
* Upgraded `@blueprintjs/core` and all other Blueprint-related packages to the latest version available.
* Switched form LibSass-based `node-sass` to Dart Sass (~`sass-embedded`~ `sass`). (I originally intended to use `sass-embedded`, which runs the Sass compiler in a Dart VM rather than using a transpiled-to-JS version, but it still has some issues which failed the builds.)
* Fixed a couple of issues related to build tooling etc. (Customizing and building Blueprint from the SCSS source isn’t officially supported, so it’s a bit of a hassle.
* Replaced all references to the Blueprint version number (in CSS class prefixes, JSX, component names and imports, …) with SCSS variables or JS constants exported by the respective packages.

   For example, instead of a CSS selector like `.bp3-input-group` we now use `.#{$bp-ns}-input-group`. `$bp-ns` is defined by Blueprint and refers to class namespace for the current major Blueprint version, for example `bp3` or `bp4`.
   
   In JSX components, we now use constants instead of using class names directly, for example when using Blueprint’s utility classes:

   ```diff
   - <span className="bp3-text-muted">Hello World!</span>
   + <span className={Classes.TEXT_MUTED}>Hello World!</span>
   ```

   Also, we’ve been recreating the HTML markup for some of Blueprint’s components in a few places to customize their appearance. I’ve tried to use the React components provided by Blueprint instead.

   This should make future major version upgrades easier. The Blueprint team aims to release major versions more frequently, approx. every 6 months (though with fewer breaking changes in each major release).

### 3) Testing
* I have created an incomplete least of core features/scenarios on the Wiki that I have used to systematically test the UI and discover regressions.
* I have also spent some time to explore automating these tests using Playwright.

### 4) Visual Changes
* Blueprint 4 comes with a new color palette.
* There are a few minor changes to the visual appearance of components, e.g. inputs have a slightly darker border. Nothing major though.

### 5) Network Diagram Embeds
* As we’re considering to permanently remove the options to create network diagram embeds and to export diagrams as .ftm files, I have removed these options from the UI.
* That way, we can quickly revert the changes if necessary, and safely remove all front and back end code related to this feature after some time.

### Followup todos:
* ~Fix network diagram embeds.~
* ~Fix invalid DOM nesting on investigation home pages.~
* After merging, upload merged translations to Transifex.
* Consider removing layerings/abstractions, e.g. `EntityManager`.

Closes #2146.

### CODE REVIEW

File: ui/craco.config.js
Comment (tillprochaska) [reply ui/craco.config.js:46]: This wouldn’t be possible with the current implementation, as `size` is a regex match. But I can still add an explicit coercion to make this more explicit.

https://github.com/alephdata/aleph/blob/1f9ea3045bfd5b7f8d43e6c6bd0b9ac12bb9ef89/ui/craco.config.js#L38

File: ui/i18n/mergeReactFtmTranslations.js
Comment (tillprochaska) [reply ui/i18n/mergeReactFtmTranslations.js:80]: No, but I almost forgot about it. I’ll delete the commit before we merge the PR.

File: docker-compose.dev.yml
Comment (tillprochaska) [reply docker-compose.dev.yml:143]: I have documented this approach in more detail here: https://github.com/tillprochaska/til/blob/master/docker.md#mounting-a-folder-from-a-container-to-the-host-system

It is a little hacky, and I’m open for suggestions/other approaches.

File: ui/Dockerfile
Comment (tillprochaska) [reply ui/Dockerfile:6]: Alpine Linux doesn’t ship with rsync and I have used it in the startup command for the UI service. There’s more info in this comment: https://github.com/alephdata/aleph/pull/2478/commits/3794f35f998091be636f2506eb985a582b1b4e36#r974287490

File: ui/src/app/App.scss
Comment (tillprochaska) [reply ui/src/app/App.scss:196]: The `/` division syntax in SCSS has been deprecated as it is ambiguous and isn’t supported anymore by Dart Sass. The alternatives are using the `math.div(x, y)` function (which is more verbose), or multiplying in case of simple fractions.

(These changes have been applied automatically using a migrator script provided by the Sass maintainers -- it prefers multiplication over `math.div()` in simple cases like this.)

Comment (tillprochaska) [reply ui/src/app/App.scss:196]: Read more: https://sass-lang.com/documentation/breaking-changes/slash-div

File: ui/src/react-ftm/utils/withTranslator.tsx
Comment (Rosencrantz) [inline ui/src/react-ftm/utils/withTranslator.tsx]: moroccan?

**Review by Rosencrantz — APPROVED** [review]
looks good

File: ui/src/components/Dashboard/Dashboard.scss
Comment (tillprochaska) [reply ui/src/components/Dashboard/Dashboard.scss:17]: This declaration is a duplicate -- it has been in the source before (a few lines below), but the duplicate declaration probably sneaked in here when I was resolving merge conflicts or something similar. I have now removed it.

https://github.com/alephdata/aleph/blob/f1a49a21ee9756714307378f1eeb8afc9c74eb7c/ui/src/components/Dashboard/Dashboard.scss#L36

**Pull request merged**
